bitcoin-core-dev-tech/2023-09/p2p-working-session.md

```diff
@@ -0,0 +1,204 @@
+  - you are changing the behavior of what we allow in the mempool.
+  - If some sends an INV or sends GETDATA - parent and child issues also complicate things. Convinced I can hide it.
+- Now I'm leaning the other way, which is not to put it in the mempool. But where? Need to put it on disk. In the mempool there are unbroadcasted txs. The code is already there. If it doesn't come back, you need to put it somewhere.
+- There are two cases to think about. It comes from your wallet or it's from RPC. What happens if the RPC comes from the wallet?
+- SendoutTx would put in your mempool. Subject to trickling mechanism will go in the ToBeINVed.
+  - What about announcing to some of your peers, wait to see it echoed back and then send to all.
+  - How bitcoin core worked until 0.5
+  - It's not trivial I think. Need to act differently for that Txs than Txs coming from elsewhere which may also be a tell. And also need a more aggressive broadcast mechanism until it is echoed back.
+  - we don't want to send it to everyone. Then we send through Tor, then you could skip all the rest and then you destroy the assumption of the attacker.
+  - if you fluff it, then someone can assume that you are the source wallet. Even if none of the behavior changes, it destroys the assumption reliability. It lowers the confidence level.
+  - chainalysis can't see where it originates
+- if you have a time-sensitive protocol, that might be an issue
+- after 20 min try the same
+  - 20 min seems enormous. Should only be a few mins. An attacker watching all the connections. If you only send it to 1 then they can't do the timing connection. It would be nice as a stretch goal, to obscure that the mechanism is being used.
+  - In addition to Tor peers, send to clearnet connections through Tor. In I2P, no source connections.
+- can you coin flip this, by coinflip
+- if you do it twice then they can tell it is yours
+- what do you think about sending out once immediately and after a short delay (seconds) do it again.
+- This happens now because we have only 2 threads for the connection manager. So it is going to take some time. Should I broadcast it to 5 unconditionally or 1 and as soon as
+- do we broadcast after we see it?
+- we broadcast like we've never seen it before. Will remove from the mempool and
+- a goal is to treat our transactions like others. By keeping it out of the mempool, that goal is achieved.
+- so we will keep out the mempool
+- and then we can think about wallet interaction
+- not going to see 0 balance, but now we will see the same balance
+- can't do this by default until we have the wallet interaction in place. And we deal with the case that someone makes a dependent tx because then we use the privacy advantages. The correction solution is a new state in the wallet for transactions that can't be spent.
```

Comment:
```suggestion
- can't do this by default until we have the wallet interaction in place. And we deal with the case that someone makes a dependent tx because then we use the privacy advantages. The correct solution is a new state in the wallet for transactions that can't be spent.
```